xdsl/dialects/onnx.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import TypeAlias, TypeVar
+
+from xdsl.dialects.builtin import AnyTensorType, SSAValue
+from xdsl.ir import (
+    Attribute,
+    Dialect,
+    OpResult,
+)
+from xdsl.irdl import (
+    Attribute,
+    IRDLOperation,
+    irdl_op_definition,
+    operand_def,
+    result_def,
+)
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+
+ArgT = TypeVar("ArgT", bound=Attribute)
+Operand: TypeAlias = SSAValue
+
+
+class ElementwiseBinOpBase(IRDLOperation, ABC):
+    """Base class for element-wise binary operations on tensors with Numpy-style broadcasting."""
+
+    lhs: Operand = operand_def()
+    rhs: Operand = operand_def()
+    res: OpResult = result_def(AnyTensorType)
```

Comment:
```suggestion
    lhs = operand_def()
    rhs = operand_def()
    res = result_def(AnyTensorType)
```

Reply:
I'm curious -- when do you prefer not to have type hints?

Reply:
We don't really have a strict style guide. When we migrated the code over to the new API I, regrettably, wrote a sed script that added the type hints. I now think they're just a bit noisy and not super helpful. The fact that you reached for a typealias to make it more explicit that the type corresponded to an operand is probably a good signal that they're confusing. I'd recommend against them whenever we use the `_def` things, except for existing files where we already have the type hints, where it might be worth keeping them for consistency.